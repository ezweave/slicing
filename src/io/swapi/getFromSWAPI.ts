import axios, { Axios, AxiosResponse } from 'axios';

export interface Getter {
  (path: string): (index: number) => Promise<AxiosResponse>
}

export const getFromSWAPI = (
  baseUrl: string = 'https://swapi.dev/api/',
  getFn: Axios['get'] = axios.get
): Getter => (
  path
) => async (
  index
) => getFn(`${baseUrl}${path}/${index}/`);
