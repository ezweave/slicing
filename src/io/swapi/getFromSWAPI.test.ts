import { writeFileSync } from 'fs';

import LarsOwenJSON from '@slicing/__fixtures__/swapi/people/lars-owen.json';

import { getFromSWAPI } from './getFromSWAPI';

describe(getFromSWAPI, () => {
  it('gets a request from SWAPI', async () => {
    const getter = jest.fn().mockResolvedValue({
      data: LarsOwenJSON,
    });
    const getPeople =
      getFromSWAPI('https://swapi.dev/api/', getter)(`people`);
    const response = await getPeople(6);
    expect(response.data).toEqual(LarsOwenJSON);
    expect(getter).toHaveBeenCalledTimes(1);
    expect(getter.mock.calls[0]).toMatchSnapshot();
  });
});
