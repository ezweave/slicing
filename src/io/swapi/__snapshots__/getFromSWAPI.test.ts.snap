// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFromSWAPI gets a request from SWAPI 1`] = `
Array [
  "https://swapi.dev/api/people/6/",
]
`;
